python/ray/experimental/channel/nccl_group.py

```diff
@@ -156,14 +215,20 @@ def send(self, value: "torch.Tensor", peer_rank: int) -> None:
         """
         if self._closed:
             raise RayChannelError("NCCL group has been destroyed.")
+
+        if isinstance(value, GPUFuture):
+            self._send_stream.synchronize()
+            value = value.wait(self._send_stream)
```

Comment:
For symmetry, it's better if we can always call `wait()`, not just if it's a `GPUFuture`. So the signature for this method should now be something like `send(fut: "DAGNodeFuture", peer_rank: int)`